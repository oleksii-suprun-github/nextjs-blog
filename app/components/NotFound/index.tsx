import Link from 'next/link';

const NotFound = ({
  link,
  title,
  description,
}: {
  link: string;
  title?: string;
  description?: string;
}) => (
  <section className="flex min-h-screen items-center justify-center bg-brand-dark-purple">
    <div className="mx-auto max-w-screen-xl px-4 py-8 lg:px-6 lg:py-16">
      <div className="mx-auto max-w-screen-sm text-center">
        <h1 className="mb-4 text-7xl font-extrabold tracking-tight text-brand-pink lg:text-9xl">
          404
        </h1>
        <p className="mb-4 text-3xl font-bold tracking-tight text-gray-900 md:text-4xl dark:text-stone-300">
          {title || 'Page Not Found'}
        </p>
        <p className="mb-4 text-lg font-light text-gray-500 dark:text-gray-400">
          {description || 'Sorry, we can`t find that publication.'}
        </p>
        <Link href={link} className="btn bg-brand-pink text-gray-900 hover:bg-brand-light-pink">
          Back to Latest Publications page
        </Link>
      </div>
    </div>
  </section>
);

export default NotFound;
