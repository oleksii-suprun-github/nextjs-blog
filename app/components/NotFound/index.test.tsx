import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import NotFound from '.';

vi.mock('next/link', () => ({
  __esModule: true,
  default: ({ href, children, ...rest }: any) => (
    <a href={href} {...rest}>
      {children}
    </a>
  ),
}));

describe('NotFound Component', () => {
  const link = '/publications';

  it('should render correctly and match snapshot', () => {
    const { asFragment } = render(<NotFound link={link} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display the correct error messages', () => {
    render(<NotFound link={link} />);

    expect(screen.getByText('404')).toBeInTheDocument();
    expect(screen.getByText('Page Not Found')).toBeInTheDocument();
    expect(screen.getByText('Sorry, we can`t find that publication.')).toBeInTheDocument();
  });

  it('should have a link back to the publications page', () => {
    render(<NotFound link={link} />);

    const backLink = screen.getByRole('link', {
      name: /Back to Latest Publications page/i,
    });
    expect(backLink).toHaveAttribute('href', link);
  });
});
