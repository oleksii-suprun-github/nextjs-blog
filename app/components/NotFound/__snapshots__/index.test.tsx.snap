// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound Component > should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <section
    class="flex min-h-screen items-center justify-center bg-brand-dark-purple"
  >
    <div
      class="mx-auto max-w-screen-xl px-4 py-8 lg:px-6 lg:py-16"
    >
      <div
        class="mx-auto max-w-screen-sm text-center"
      >
        <h1
          class="mb-4 text-7xl font-extrabold tracking-tight text-brand-pink lg:text-9xl"
        >
          404
        </h1>
        <p
          class="dark: mb-4 text-3xl font-bold tracking-tight text-gray-900 md:text-4xl"
        >
          Page Not Found
        </p>
        <p
          class="mb-4 text-lg font-light text-gray-500 dark:text-gray-400"
        >
          Sorry, we can\`t find that publication.
        </p>
        <a
          class="btn bg-brand-pink text-gray-900 hover:bg-brand-light-pink"
          href="/publications"
        >
          Back to Latest Publications page
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
